tests/stub/routing.py

```diff
@@ -2008,6 +2009,12 @@ def test_should_accept_routing_table_without_writers_and_then_rediscover(self):
                                  database=self.get_db())
         sequences = []
         self._routingServer1.done()
+        try:
+            driver.verifyConnectivity()
+        except types.DriverError:
+            # make sure the driver noticed that its old connection to
+            # _routingServer1 is dead
```

Comment:
I can see the comment, but I am not sure why the driver needs to notice this... Could you explain please? (I know this test currently relies on the driver fetching RT in the `verify` method, which is suboptimal.)

Reply:
Tried it again, works. I think it was a left-over from when the Python driver implemented the `verifyConnectivity` method in a slightly different way :shrug:

Reply:
@injectives turns out it was needed indeed. Must have missed it when testing again. The reason is that the Python driver keeps the connection to the router alive in the pool. Then `session.run("RETURN 1 as n").consume()` will cause the driver to try to re-fetch the RT because it's not knowing any writers. Since this in an auto-commit tx, there is no retry. The pool connection will fail, the tx, will fail, the test will fail.